src/pages/scientificServices/pipelines/views/TableFilters.tsx

```diff
@@ -0,0 +1,207 @@
+import { ButtonPrimary, Select } from '@terra-ui-packages/components';
+import React from 'react';
+import { DelayedSearchInput } from 'src/components/input';
+import { Pipeline, PipelineRunStatus } from 'src/libs/ajax/teaspoons/teaspoons-models';
+
+export interface FilterValues {
+  description?: string;
+  jobId?: string;
+  status?: PipelineRunStatus;
+  pipelineName?: string;
+}
+
+interface TableFiltersProps {
+  filters: FilterValues;
+  onFilterChange: (filters: FilterValues) => void;
+  pipelinesList: Pipeline[];
+}
+
+export const TableFilters: React.FC<TableFiltersProps> = ({ filters, onFilterChange, pipelinesList }) => {
+  const handleInputChange = (field: keyof FilterValues, value: string) => {
+    onFilterChange({
+      ...filters,
+      [field]: value || undefined,
+    });
+  };
+
+  const handleClearFilters = () => {
+    onFilterChange({});
+  };
+
+  const hasActiveFilters = Object.values(filters).some((value) => value !== undefined && value !== '');
+
+  return (
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        gap: '1rem',
+        backgroundColor: '#eff0f1',
+        border: '1px solid #d7d9dc',
+        borderRadius: '0.25rem',
+        padding: '0.75rem 0.75rem',
+      }}
+    >
+      <div style={{ display: 'flex', gap: '1rem', flexWrap: 'wrap', alignItems: 'flex-end' }}>
+        <DescriptionFilterControl filters={filters} handleInputChange={handleInputChange} />
+        <JobIdFilterControl filters={filters} handleInputChange={handleInputChange} />
+        <StatusFilterControl filters={filters} handleInputChange={handleInputChange} />
+        <PipelineFilterControl filters={filters} handleInputChange={handleInputChange} pipelinesList={pipelinesList} />
+
+        <div
+          style={{
+            display: 'flex',
+            flex: '0 0 auto',
+            justifyContent: 'center',
+          }}
+        >
+          <ButtonPrimary disabled={!hasActiveFilters} onClick={handleClearFilters}>
+            Clear
+          </ButtonPrimary>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const DescriptionFilterControl = ({
+  filters,
+  handleInputChange,
+}: {
+  filters: FilterValues;
+  handleInputChange: (field: keyof FilterValues, value: string) => void;
+}) => {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'column',
+        gap: '0.25rem',
+        minWidth: '200px',
+        flex: '1 1 25%',
+      }}
+    >
+      <div style={{ fontWeight: 600, fontSize: '14px' }}>Description</div>
+      <DelayedSearchInput
```

Comment:
these inputs are debounced to avoid hitting the backend for every single keystroke in the text input